### PR TITLE
Adding SlicerMorph Extension

### PR DESCRIPTION
**Pull request by smrolfe**
# Submitting a new extension
This pull request adds the SlicerMorph extension for the first time.

- [x] Extension has a reasonable name (not too general, not too narrow, suggests what the extension is for)
- [x] Repository name is Slicer+ExtensionName
- [x] Extension description summarizes in 1-2 sentences what the extension is usable (should be understandable for non-experts)
- [x] Extension URL and revision (scmurl, scmrevision) is correct, consider using a branch name (master, release, ...) instead of a specific git has to avoid re-submitting pull request whenever the extension is updated
- [x] Extension icon URL is correct
- [x] Screenshot URLs (screenshoturls) are correct, contains at least one
- [x] Homepage URL points to valid webpage containing the following:
  - [x] Extension name
  - [x] Short description: 1-2 sentences, which summarizes what the extension is usable for
  - [x] At least one nice, informative image, that illustrates what the extension can do. It may be a screenshot.
  - [x] Description of contained modules: at one sentence for each module
  - [x] Tutorial: step-by-step description of at least the most typical use case, include a few screenshots, provide download links to sample input data set
  - [x] Publication: link to publication and/or to PubMed reference (if available)
  - [x] License: We suggest you use a permissive license that includes patent and contribution clauses.  This will help protect developers and ensure the code remains freely available.  We suggest you use the [Slicer License](https://github.com/Slicer/Slicer/blob/master/License.txt) or the [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0). Always mention in your README file the license you have chosen.  If you choose a different license, explain why to the extension maintainers. Depending on the license we may not be able to host your work.
Read [here](https://opensource.guide/legal/#which-open-source-license-is-appropriate-for-my-project) to learn more about licenses.

### CODE REVIEW

**Comment by jcfr** [comment]
To make sure users can find your extension, understand what it is intended for and how to use it, please complete the checklist below. You do not need to complete all the item by the time you submit the pull request, but most likely the changes will only be merged if all the tasks are done.

- [ ] Extension has a reasonable name (not too general, not too narrow, suggests what the extension is for)
- [ ] Repository name is Slicer+ExtensionName
- [ ] Extension description summarizes in 1-2 sentences what the extension is usable (should be understandable for non-experts)
- [ ] Extension URL and revision (scmurl, scmrevision) is correct, consider using a branch name (master, release, ...) instead of a specific git has to avoid re-submitting pull request whenever the extension is updated
- [ ] Extension icon URL is correct
- [ ] Screenshot URLs (screenshoturls) are correct, contains at least one
- [ ] Homepage URL points to valid webpage containing the following:
  - [ ] Extension name
  - [ ] Short description: 1-2 sentences, which summarizes what the extension is usable for
  - [ ] At least one nice, informative image, that illustrates what the extension can do. It may be a screenshot.
  - [ ] Description of contained modules: at one sentence for each module
  - [ ] Tutorial: step-by-step description of at least the most typical use case, include a few screenshots, provide download links to sample input data set
  - [ ] Publication: link to publication and/or to PubMed reference (if available)
  - [ ] License: We suggest you use a permissive license that includes patent and contribution clauses.  This will help protect developers and ensure the code remains freely available.  We suggest you use the [Slicer License](https://github.com/Slicer/Slicer/blob/master/License.txt) or the [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0). Always mention in your README file the license you have chosen.  If you choose a different license, explain why to the extension maintainers. Depending on the license we may not be able to host your work.
Read [here](https://opensource.guide/legal/#which-open-source-license-is-appropriate-for-my-project) to learn more about licenses.

See more information about the submission process on the Slicer wiki: https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/BuildTestPackageDistributeExtensions

If you have any questions or comments then please describe them here.

File: SlicerMorphExtension.s4ext [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# Firstname1 Lastname1 ([SubOrg1, ]Org1), Firstname2 Lastname2 ([SubOrg2, ]Org2)
+# For example: Jane Roe (Superware), John Doe (Lab1, Nowhere), Joe Bloggs (Noware)
+contributors Sara Rolfe (UW), Murat Maga (SCRI)
+
+# Match category in the xml description of the module (where it shows up in Modules menu)
+category    SlicerMorph
+
+# url to icon (png, size 128x128 pixels)
+iconurl     https://github.com/SlicerMorph/SlicerMorph/blob/master/SlicerMorph_Color.png
```
Comment (jcfr) [inline SlicerMorphExtension.s4ext]: -> https://raw.githubusercontent.com/SlicerMorph/SlicerMorph/master/SlicerMorph_Color.png

**Comment by smrolfe** [comment]
Thanks @jcfr for these great comments! I have made all the changes suggested.

Can you tell me the best way to address the checkmark list? Should I copy and past here to update?

**Comment by lassoan** [comment]
You should have kept the checklist (it was part of the template). I've tried to make the pull [request template](https://github.com/Slicer/ExtensionsIndex/blob/master/.github/PULL_REQUEST_TEMPLATE.md) a bit more clear, but please let us know if you have any suggestions for improving it.

Anyway, I moved the checklist to the main post now so you should be able to edit it.

**Comment by jcfr** [comment]
@smrolfe  Thanks for updating. And thanks to @lassoan , you should now be able to update the checklist.

**Comment by smrolfe** [comment]
Thanks @lassoan and @jcfr . The new instructions on the template are more clear.  I have completed the prerequisite items and updated the checklist.

Could you also give me an idea how long it usually takes for an extension to be available after it is reviewed?

**Comment by jcfr** [comment]
After we integrate the PR, it is available next day with the nightly.

**Comment by smrolfe** [comment]
Hi @jcfr I just checked and didn't see SlicerMorph in the extension manager. Can you let me know if I should have seen it by now, or if the index has not been updated yet?

**Comment by jcfr** [comment]
You can see the submission result on the dashboard. This link filter the list of result.

See http://slicer.cdash.org/index.php?project=SlicerPreview&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=SlicerMorph

**Comment by jcfr** [comment]
Building the project locally should help reproduce the issue. For example, a parenthesis is missing here: https://github.com/SlicerMorph/SlicerMorph/blob/1f7e2712d4e4155e20700dd5048ea28f929468a2/ConvertMorphologikaLandmarks/CMakeLists.txt#L2

**Comment by smrolfe** [comment]
@jcfr Thanks for your help!

**Comment by smrolfe** [comment]
The extension is now showing up, but the icon is not displaying in the Extension Manager. The [link ](http://slicer.kitware.com/midas3/slicerappstore/extension/SlicerMorph_Color.png)in the index file is working but the [icon URL](http://slicer.kitware.com/midas3/slicerappstore/extension/SlicerMorph_Color.png) in the extension manager doesn't exist. Could there be a problem with the image file that would cause this?

**Comment by jcfr** [comment]
The meta-data in the CMakeLists.txt are also expected to be set appropriately. This should be fixed in https://github.com/SlicerMorph/SlicerMorph/pull/3

**Comment by smrolfe** [comment]
Thanks for your explanation, I have merged your changes.

**Pull request merged**